src/timemanager.cxx

```diff
@@ -268,6 +268,7 @@ void TimeManager::process(Buffers* buffers)
 		// write new beat to UI (bar info currently not used)
 		EventTimeBarBeat e( barCounter, beatCounter );
 		writeToGuiRingbuffer( &e );
+		writeToDspRingbuffer( &e );
```

Comment:
Is this the right way to queue the event, in both ringbuffers?

Reply:
No, you should only ever need to enqueue to one of the ringbuffers. Keep in mind they're both SingleProducer, SingleConsumer (SP/SC for short) ringbuffers. It is not safe to enqueue to both, from a threading perspective.